pallets/pallet-lottery/src/staking/mod.rs

```diff
@@ -219,9 +219,17 @@ impl<T: Config> Pallet<T> {
     ) -> DispatchResult {
         log::trace!(function_name!());
         // preconditions
-        if Self::surplus_funds().is_zero() {
-            return Err(Error::<T>::PotBalanceTooLow.into());
+        // funds eligible for staking:
+        // - newly deposited funds > min_deposit
+        // - unstaked-but-not-needed-for-withdrawals funds > min_deposit
+        if amount < Self::min_deposit() {
+            return Err(Error::<T>::DepositBelowMinAmount.into());
         }
+        if amount > Self::surplus_funds() && amount > Self::unlocked_unstaking_funds() {
```

Comment:
should this be or no 
```rust
if amount > Self::surplus_funds().saturating_add(Self::unlocked_unstaking_funds()) {
```

Reply:
No, these are two separate conditions being checked here, see the comment starting L222